### PR TITLE
Fixup some tests and code

### PR DESCRIPTION
**Pull request by jasonLaster**
@ianmstew just went through your amazing PR and started fixing some tests and code. Feel free to take or reject anything here

### CODE REVIEW

File: src/monitor-view-events.js [hunk 1]
```diff
@@ -9,25 +9,32 @@ import { triggerMethodOn, triggerMethodMany } from './trigger-method';
 // whenever a rendered view is attached or an attached view is rendered.
 function MonitorViewEvents(view) {
   if (view._areViewEventsMonitored) { return; }
+
+  if (!view._getImmediateChildren) {
+    return;
+  }
```
Comment (jasonLaster) [inline src/monitor-view-events.js]: was getting errors where a view didn't have this method. Is this the right fix?

File: src/monitor-view-events.js [hunk 2]
```diff
@@ -9,25 +9,32 @@ import { triggerMethodOn, triggerMethodMany } from './trigger-method';
...
 
   function handleBeforeDetach() {
-    triggerMethodMany(view._getImmediateChildren(), 'before:detach');
+    triggerMethodMany(view._getImmediateChildren(), view, 'before:detach');
   }
 
   function handleDetach() {
     view._isAttached = false;
-    triggerMethodMany(view._getImmediateChildren(), 'detach');
+    triggerMethodMany(view._getImmediateChildren(), view, 'detach');
```
Comment (jasonLaster) [inline src/monitor-view-events.js]: needed to pass the source along

File: src/region.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-import isNodeAttached    from './utils/isNodeAttached';
-import MarionetteObject  from './object';
-import MarionetteError   from './error';
+import _ from 'underscore';
+import Backbone from 'backbone';
+import isNodeAttached from './utils/isNodeAttached';
+import MarionetteObject from './object';
+import MarionetteError from './error';
 import MonitorViewEvents from './monitor-view-events';
+import getValue from './utils/getValue';
```
Comment (jasonLaster) [inline src/region.js]: aligning froms is for chumps

File: src/region.js [hunk 4]
```diff
@@ -111,12 +112,12 @@ const Region = MarionetteObject.extend({
 
     // Attach the view
     if (isAttachedRegion && shouldTriggerAttach) {
-      triggerMethodOn(view, this, 'before:attach');
+      triggerMethodOn(view, 'before:attach', view);
```
Comment (jasonLaster) [inline src/region.js]: oops, needed to re-arrange the args

File: src/region.js [hunk 5]
```diff
@@ -125,7 +126,7 @@ const Region = MarionetteObject.extend({
 
   _ensureElement: function() {
     if (!_.isObject(this.el)) {
-      this.$el = Backbone.$(this.el, this.getValue(this.getOption('parentEl')));
+      this.$el = this.getEl(this.el);
```
Comment (jasonLaster) [inline src/region.js]: re-added `getEl`

File: test/unit/collection-view.filter.spec.js [hunk 6]
```diff
@@ -71,7 +71,7 @@ describe('collection view - filter', function() {
       this.collection.add(this.passModel);
       this.collection.add(this.failModel);
       this.collectionView = new this.CollectionView();
-      this.sinon.spy(this.collectionView, 'removeChildView');
+      this.sinon.spy(this.collectionView, '_removeChildView');
```
Comment (jasonLaster) [inline test/unit/collection-view.filter.spec.js]: privatize the public stuff

File: test/unit/get-nested-views.spec.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-describe('getNestedView', function() {
+xdescribe('getNestedView', function() {
```
Comment (jasonLaster) [inline test/unit/get-nested-views.spec.js]: well, not sure if we want these any more...

File: src/monitor-view-events.js (hunk 1 above)
Comment (paulfalgout) [inline src/monitor-view-events.js]: How does a view get to this point if it doesn't have this function?  This doesn't feel right to me.  I would expect that MonitorViewEvents would only be applied to views that have it.

File: test/unit/get-nested-views.spec.js (hunk 7 above)
Comment (paulfalgout) [inline test/unit/get-nested-views.spec.js]: FYI you can do `describe.skip(`  I learned that from @brianmann

**Comment by ianmstew** [comment]
This is wonderful, Jason.  Thanks for taking a thorough look and for fixes!!  I'll merge this in before pushing again.

File: src/monitor-view-events.js (hunk 1 above)
Comment (jasonLaster) [inline src/monitor-view-events.js]: probably

File: src/monitor-view-events.js (hunk 2 above)
Comment (ianmstew) [inline src/monitor-view-events.js]: Ah yeah, I saw that... then I second guessed the pattern.

File: src/region.js (hunk 3 above)
Comment (ianmstew) [inline src/region.js]: Haha :p

File: src/region.js (hunk 4 above)
Comment (ianmstew) [inline src/region.js]: :frowning:

File: src/region.js (hunk 5 above)
Comment (ianmstew) [inline src/region.js]: Thanks, did this locally too, I'll address any conflict in a manual merge.

File: test/unit/collection-view.filter.spec.js (hunk 6 above)
Comment (ianmstew) [inline test/unit/collection-view.filter.spec.js]: Haha

File: src/monitor-view-events.js (hunk 1 above)
Comment (ianmstew) [inline src/monitor-view-events.js]: I'm guessing Backbone.View :/

Comment (paulfalgout) [inline src/monitor-view-events.js]: Ah.. because it's added in the region's show in addition to the view constructor.. that makes sense.

**Comment by ianmstew** [comment]
Manually merged into #2850

File: test/unit/get-nested-views.spec.js (hunk 7 above)
Comment (ianmstew) [inline test/unit/get-nested-views.spec.js]: Went with a wholesale removal of the unused specs.

**Pull request closed without merging**